internal/tools/applications.go

```diff
@@ -272,6 +272,120 @@ func GetCreateM2MAppTool() (mcp.Tool, server.ToolHandlerFunc) {
 	return mobileAppTool, mobileAppToolImpl
 }
 
+func GetCreateReactAppTool() (mcp.Tool, server.ToolHandlerFunc) {
+	productName := config.GetProductName()
+	client, err := asgardeo.GetClientInstance(context.Background())
+
+	if err != nil {
+		log.Printf("Error initializing client instance: %v", err)
+	}
+
+	reactAppTool := mcp.NewTool("create_react_app",
+		mcp.WithDescription(fmt.Sprintf("Create a new React Application in %s", productName)),
+		mcp.WithString("application_name", mcp.Description("Name of the application"), mcp.Required()),
+		mcp.WithString("redirect_url", mcp.Description("Redirect URL of the application"), mcp.Required()),
+	)
+
+	reactAppToolImpl := func(ctx context.Context, req mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+		appName := req.Params.Arguments["application_name"].(string)
+		redirectURL := req.Params.Arguments["redirect_url"].(string)
+
+		reactApp, err := client.Application.CreateSinglePageApp(ctx, appName, redirectURL)
```

Comment:
don't we need to send the `templateId: "react-application"` to show this with the React template in console?